### PR TITLE
feat: [ANDROAPP-3479] show/hide filters from webapp

### PR DESCRIPTION
**Pull request by JaimeToca**
## Description
Please include a summary of the change and include the related jira issue if it exists.

[ jira issue ](https://jira.dhis2.org/browse/

## Solution description
If this PR is a fix include a brief description on how the issue is solved.
## Covered unit test cases
Describe the tests that you ran to verify your changes.
## Where did you test this issue?
- [ ] Smartphone Emulator
- [ ] Tablet Emulator
- [ ] Smartphone
- [ ] Tablet
## Which Android versions did you test this issue?
- [ ] 4.4
- [ ] 5.X - 6.X
- [ ] 7.X
- [ ] 8.X
- [ ] 9.X - 10.X
- [ ] Other
## Checklist
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code

### CODE REVIEW

File: app/src/main/java/org/dhis2/data/filter/FilterRepository.kt [hunk 1]
```diff
@@ -258,313 +264,413 @@ class FilterRepository @Inject constructor(
...
+        /* val globalTrackerFiltersWebApp: Map<ProgramFilter, FilterConfig> = mapOf() // SDK global call
+         val globalTrackerFiltersWebAppKey =
+             globalTrackerFiltersWebApp.filterValues { it.filter() }.keys.toList()
+         val filtersToShow = defaultFilters.filter { globalTrackerFiltersWebAppKey.contains(it.key) }
+
+         if (filtersToShow.isEmpty()) return mutableListOf() */
+
+        //return filtersToShow.values.toList()
+
+        return defaultFilters.values.toList()
```
Comment (JaimeToca) [inline app/src/main/java/org/dhis2/data/filter/FilterRepository.kt:284]: This still needs to be define

**Comment by JaimeToca** [comment]
Wait for the code review until there are no blockers anymore

**Comment by JaimeToca** [comment]
@andresmr @Balcan wait for review I am enabling unit test in the filter repository class due to android dependencies

File: app/src/main/java/org/dhis2/data/filter/GetFiltersApplyingWebAppConfig.kt [hunk 2]
```diff
@@ -0,0 +1,17 @@
+package org.dhis2.data.filter
+
+import org.dhis2.utils.filters.FilterItem
+import org.hisp.dhis.android.core.settings.FilterSetting
+
+class GetFiltersApplyingWebAppConfig {
+
+    fun <T> execute(
```
Comment (andresmr) [inline app/src/main/java/org/dhis2/data/filter/GetFiltersApplyingWebAppConfig.kt:8]: If it is only used in FilterRepository and it's test, why is not a private method of it?

**Review by andresmr — APPROVED** [review]

**Review by Balcan — APPROVED** [review]

**Pull request merged**